modules/seastate/src/SeaState_Input.f90

```diff
@@ -759,122 +677,93 @@ subroutine SeaStateInput_ProcessInitData( InitInp, p, InputFileData, ErrStat, Er
-
-         InputFileData%Waves%WavePkShp = WavePkShpDefault ( InputFileData%Waves%WaveHs, InputFileData%Waves%WaveTp )
-
-      else                                   ! The input must have been specified numerically.
-
-         read (InputFileData%Waves%WavePkShpChr,*,IOSTAT=IOS)  InputFileData%Waves%WavePkShp
-            call CheckIOS ( IOS, "", 'WavePkShp', NumType, ErrStat2, ErrMsg2 )
-            call SetErrStat(ErrStat2, ErrMsg2,ErrStat,ErrMsg,RoutineName)
-            if ( ErrStat >= AbortErrLev ) return
-
-         if ( ( InputFileData%Waves%WavePkShp < 1.0 ) .OR. ( InputFileData%Waves%WavePkShp > 7.0 ) )  then
-            call SetErrStat( ErrID_Fatal,'WavePkShp must be greater than or equal to 1 and less than or equal to 7.',ErrStat,ErrMsg,RoutineName)
-            return
-         end if
-
-      end if
+   if ( InputFileData%Waves%WaveTp <= 0.0 )  then
+      call SetErrStat( ErrID_Fatal,'WaveTp must be greater than zero.',ErrStat,ErrMsg,RoutineName)
+      return
+   end if
 
-   else
 
-      InputFileData%Waves%WavePkShp = 1.0
 
+       ! WavePkShp - Peak shape parameter
+   if ( ( InputFileData%Waves%WavePkShp < 1.0 ) .OR. ( InputFileData%Waves%WavePkShp > 7.0 ) )  then
+      call SetErrStat( ErrID_Fatal,'WavePkShp must be greater than or equal to 1 and less than or equal to 7.',ErrStat,ErrMsg,RoutineName)
+      return
    end if
 
 
       ! WvLowCOff and WvHiCOff - Wave Cut-off frequency
 
-   if ( InputFileData%Waves%WvLowCOff < 0 ) then
+   if ( InputFileData%WvLowCOff < 0 ) then
       call SetErrStat( ErrID_Fatal,'WvLowCOff must be greater than or equal to zero.',ErrStat,ErrMsg,RoutineName)
       return
    end if
 
       ! Threshold upper cut-off based on sampling rate
    if ( EqualRealNos(InputFileData%Waves%WaveDT, 0.0_DbKi) ) then
-      InputFileData%Waves%WvHiCOff = 10000.0;  ! This is not going to be used because WaveDT is zero.
+      InputFileData%WvHiCOff = 10000.0;  ! This is not going to be used because WaveDT is zero.
    else
       TmpFreq = REAL( Pi/InputFileData%Waves%WaveDT,SiKi)
-      if ( InputFileData%Waves%WvHiCOff > TmpFreq ) then
-         InputFileData%Waves%WvHiCOff =  TmpFreq
-         call SetErrStat( ErrID_Info,'WvLowCOff adjusted to '//trim(num2lstr(TmpFreq))//' rad/s, based on WaveDT.',ErrStat,ErrMsg,RoutineName)
+      if ( InputFileData%WvHiCOff > TmpFreq ) then
+         InputFileData%WvHiCOff =  TmpFreq
+         call SetErrStat( ErrID_Info,'WvHiCOff adjusted to '//trim(num2lstr(TmpFreq))//' rad/s, based on WaveDT.',ErrStat,ErrMsg,RoutineName)
       end if
    end if
 
-   if (InputFileData%Waves%WaveMod > 2 .and. InputFileData%Waves%WaveMod /= 6) then
-      if ( InputFileData%Waves%WvLowCOff >= InputFileData%Waves%WvHiCOff ) then
+   !bjj: do we even need this check on WaveMod? Even if it's not being used, we can check that low < Hi, right?
```

Comment:
@bjonkman Agreed. There's no harm in requiring `WvHiCOff > WvLowCOff` even if they are not used. We can remove the `WaveMod` checks here.

Reply:
As a user I prefer if the sanity checks are only done for the options that are activated. I like to put garbage in the inputs I don't use, to make it visually clear that I'm not using them.

Reply:
If we are to only perform the checks for the relevant `WaveMod` options, I think the ones that should be included are 2 (`WaveMod_JONSWAP`), 3 (`WaveMod_WhiteNoise`), 4 (`WaveMod_UserSpctrm`), 5 (`WaveMod_ExtElev`), and 7 (`WaveMod_UserFreq`) if I'm not mistaken. Not sure why 2 (JONSWAP) was not included in the original check. We might also want to overwrite `WvLowCOff` and `WvHiCOff` for the two regular-wave options in the code here just to make sure we are not cutting out that one wave component.

Reply:
Addressed in 67c77c: The check has been updated for relevant WaveMod options. For WaveMod options that are supposed to ignore these values, I set them to -HUGE() and +HUGE() so that when they ARE used in HydroDyn, they allow all frequencies.